tests/unit/unpivot-step-form.spec.ts

```diff
@@ -0,0 +1,260 @@
+      await Vue.nextTick();
+      const errors = wrapper.vm.$data.errors.map((err: ValidationError) => ({
+        keyword: err.keyword,
+        dataPath: err.dataPath,
+        message: err.message,
+      }));
+      expect(errors).to.eql([
+        {
+          keyword: 'columnNameConflict',
+          dataPath: '.unpivot_column_name',
+          message: 'Column name columnA is used at least twice but should be unique',
+        },
+      ]);
+    });
+  });
+
+  it('should report errors when the value_column_name value is already used', async () => {
+    const store = setupStore({
+      dataset: {
+        headers: [{ name: 'columnA' }, { name: 'columnB' }, { name: 'columnC' }],
+        data: [],
+      },
+    });
+    const wrapper = mount(UnpivotStepForm, { store, localVue });
+    wrapper.setData({
+      editedStep: {
+        name: 'unpivot',
+        keep: ['columnA'],
+        unpivot: ['columnB', 'columnC'],
+        unpivot_column_name: 'foo',
+        value_column_name: 'columnB',
+        dropna: true,
+      },
+    });
+    wrapper.find('.widget-form-action__button--validate').trigger('click');
+    await Vue.nextTick();
+    const errors = wrapper.vm.$data.errors.map((err: ValidationError) => ({
+      keyword: err.keyword,
+      dataPath: err.dataPath,
+      message: err.message,
+    }));
+    expect(errors).to.eql([
+      {
+        keyword: 'columnNameConflict',
+        dataPath: '.value_column_name',
+        message: 'Column name columnB is used at least twice but should be unique',
+      },
+    ]);
+  });
+
+  it('should validate and emit "formSaved" when submitted data is valid', async () => {
+    const wrapper = mount(UnpivotStepForm, {
+      store: emptyStore,
+      localVue,
+      propsData: {
+        initialStepValue: {
+          name: 'unpivot',
+          keep: ['columnA', 'columnB'],
+          unpivot: ['columnC'],
+          unpivot_column_name: 'foo',
+          value_column_name: 'bar',
+          dropna: true,
+        },
+      },
+    });
+    wrapper.find('.widget-form-action__button--validate').trigger('click');
+    await Vue.nextTick();
+    expect(wrapper.vm.$data.errors).to.be.null;
+    expect(wrapper.emitted()).to.eql({
+      formSaved: [
+        [
+          {
+            name: 'unpivot',
+            keep: ['columnA', 'columnB'],
+            unpivot: ['columnC'],
+            unpivot_column_name: 'foo',
+            value_column_name: 'bar',
+            dropna: true,
+          },
+        ],
+      ],
+    });
+  });
+
+  it('should emit "cancel" event when edition is cancelled', async () => {
+    const wrapper = mount(UnpivotStepForm, { store: emptyStore, localVue });
+    wrapper.find('.widget-form-action__button--cancel').trigger('click');
+    await Vue.nextTick();
+    expect(wrapper.emitted()).to.eql({
+      cancel: [[]],
+    });
+  });
+
+  it('should reset selectedStepIndex correctly on cancel depending on isStepCreation', () => {
```

Comment:
I'm not sure about the meaning of this test, which User Case doest that represent ?

Reply:
It is to test that the focus is made on the correct step after the form is closed. But this is the kind of test that gets duplicated in every form due to legacy, but it should be tested once on the StepForm base class. As we have not refactored those tests yet, I propose to let this test here for the moment

Reply:
👍